devtools/server/actors/replay/connection.js

```diff
@@ -1655,6 +1656,71 @@ function sendChannelResponseStart(channel, fromCache, fromServiceWorker) {
   });
 }
 
+//
+// Handle request redirects.
+//
+const SINK_CLASS_DESCRIPTION = "Replay Networking Event Sink In Parent";
+const SINK_CLASS_ID = Components.ID("{e9caba93-270b-486c-95e1-894864b40ac2}");
+const SINK_CONTRACT_ID = "@mozilla.org/network/monitor/channeleventsink;1";
+const SINK_CATEGORY_NAME = "net-channel-event-sinks";
+class ReplayChannelEventSinkParent {
+  asyncOnChannelRedirect(oldChannel, newChannel, flags, callback) {
```

Comment:
I'm thinking we should validate the flags here so something somewhere can make sure this is an internal redirect